core-processor/src/main/java/io/micronaut/inject/writer/BeanDefinitionWriter.java

```diff
@@ -2087,6 +2099,32 @@ private int pushGetValueForPathCall(GeneratorAdapter injectMethodVisitor, ClassE
         return optionalInstanceIndex;
     }
 
+    private boolean pushValueBypassingBeanContext(GeneratorAdapter writer, ClassElement type) {
+        // Used in instantiate and inject methods
+        if (type.isAssignable(BeanResolutionContext.class)) {
+            writer.loadArg(INSTANTIATE_METHOD_BEAN_RESOLUTION_CONTEXT_PARAM);
+            return true;
+        }
+        if (type.isAssignable(BeanContext.class)) {
+            writer.loadArg(INSTANTIATE_METHOD_BEAN_CONTEXT_PARAM);
+            return true;
+        }
+        if (visitorContext.getClassElement(ConversionService.class).orElseThrow().isAssignable(type)) {
```

Comment:
`.isAssignable(type)` would also be `true` for `Object`

Reply:
Hmm right

Reply:
Fixed